instagram/src/components/Authentication/withAuthenticate.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const withAuthenticate = FirstComponent => SecondComponent => props => {
+    // return class extends React.Component {
+    //     constructor(props) {
+    //         super(props);
+
+    //         this.state = {
+    //             loggedIn: false
+    //         }
+    //     }
+
+    //     componentDidMount() {
+    //         if (localStorage.getItem('username')) {
+    //             this.setState({
+    //                 loggedIn: true
+    //             })
+    //         }
+    //     }
+
+    //     render() {
+
+    //         if (this.state.loggedIn === true) {
+    //             return <FirstComponent {...props}/>
+    //         } else {
+    //             return <SecondComponent {...props}/>
+    //         }
+    //     }
+    
+    // }
+
+    if (localStorage.getItem('username') === 'fozcat' && localStorage.getItem('password') === 'test') {
+        return <FirstComponent {...props}/>
+    } else {
+        return <SecondComponent {...props}/>
+    }
+}
+
```

Comment:
Well done!

Reply:
Please don't hardcode the `username` and `password` like this, because it is impossible for me to test the app without poking into your code to find them. I thought the app was not working.
If you want to hardcode them, at least give a notification to user that you're doing so :)